provider/terraform/custom_code.rb

```diff
@@ -66,8 +66,27 @@ class Examples < Api::Object
       # }
       attr_reader :primary_resource_id
 
-      # vars is a Hash from template variable names to output variable names
+      # vars is a Hash from template variable names to output variable names.
+      # It will use the provided value as a prefix for generated tests, and
+      # insert it into the docs verbatim.
       attr_reader :vars
+      # Some variables need to hold special values during tests, and cannot
+      # be autodiscovered for open-in-cloud-shell.  For instance, org_id
+      # needs to be the correct value during integration tests, or else
+      # org tests cannot pass. Other examples include an existing project_id,
+      # zone, service account name, etc.
+      #
+      # test_env_vars is a Hash from template variable names to one of the
+      # following symbols:
+      #  - :PROJECT_NAME
+      #  - :CREDENTIALS
+      #  - :REGION
+      #  - :ORG_ID
+      #  - :ORG_TARGET
+      #  - :BILLING_ACCT
+      #  - :SERVICE_ACCT
+      # This list corresponds to https://bit.ly/2PKfwxP (provider_test.go:150).
```

Comment:
Let's use a full link instead of bit.ly, there's no reason to add a dependency on it to our codebase.

Reply:
SGTM!